tensorflow/core/kernels/eigen_spatial_convolutions-inl.h

```diff
@@ -347,13 +378,131 @@ class TensorContractionInputMapper<
     if (nonStandardPatches()) {
       return packetWithPossibleZero(patchId, rowIndex, colIndex, otherIndex);
     }
-    return loadPacketStandard(patchId, rowIndex, colIndex, otherIndex);
+    typedef decltype(m_impl) TensorEvaluatorT;
+    return loadPacketStandard<Packet, TensorEvaluatorT>(patchId, rowIndex,
+                                                        colIndex, otherIndex);
+  }
+
+  // Helper function to load a 'partial' packet - this is the single column
+  // part of a packet that is split across two columns.
+  // If required, this function is called from
+  // loadPacketStandardFromTwoColumns() when the packet type supports masked
+  // load and when the partial packet load is available in the TensorEvaluator.
+  EIGEN_DEVICE_FUNC
+  EIGEN_ALWAYS_INLINE Packet loadPartialPacketStandard(
+      Index patchId, Index rowIndex, Index colIndex, Index otherIndex,
+      Index patchIdPart, const Index spanPart[], const Index patchOffsetsPart[],
+      Index colOffsetPart) const {
```

Comment:
It seems there is no real "semantic difference" between `patchOffsetsPart` and `patchOffsets`, I tried to find one because of different names, but it seems they are the "same".

Also I'd just pass around `colOffsets[]` and `eigen_asset` where appropriate that `colOffsets[0] == colOffsets[1]` or `colOffsets[0] + 1 == colOffsets[1]`, also to reduce the number of unique variable names.

Reply:
Let me try to reduce the variable names.
I had created variable names with suffix `Part` simply to indicate that the variable is intended to be associated with partial packet. My reasoning was that it would keep things segregated. But as you point out for `patchOffsetsPart` and `patchOffsets`, there are cases when they actually are the same.